.github/workflows/pr-close.yml

```diff
@@ -0,0 +1,31 @@
+name: delete preview on PR close
+
+# only trigger on pull request closed events
+on:
+  pull_request:
+    types: [closed]
+
+jobs:
+  delete_preview:
+    runs-on: ubuntu-latest
+    env:
+      PR_PATH: pull/${{github.event.number}}
+    steps:
+      - name: make empty dir
+        run: mkdir public
+
+      - name: delete folder
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./public
+          destination_dir: ${{ env.PR_PATH }}
+
+      - name: Comment on PR
+        uses: hasura/comment-progress@v2.2.0
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          repository: ${{ github.repository }}
+          number: ${{ github.event.number }}
+          id: deploy-preview
+          message: "🪓 PR closed, deleted preview at https://github.com/${{ github.repository }}/tree/gh-pages/${{ env.PR_PATH }}/"
```

Comment:
Suggested edits to the removal message so that it looks closer to the preview message so it preserves consistency making it easier to find when scanning for it, but updated appropriately for the deleted/closed status:
```suggestion
          message: "### Deployment Link\nThis PR has been merged or closed, the created build preview has been deleted.  (_Last Updated: ${{ github.event.created_at }}_)\n\nReview Link:\n:axe:&ensp;\~_the preview link has been removed_\~"
```
\
\
\
Visual representation of how these edits would look:
<img width="915" alt="new-action-comment-removed" src="https://user-images.githubusercontent.com/1355485/226867402-a8a0022d-3269-4f54-803d-e1a6eb83d4ee.png">